### PR TITLE
add bmi2 to generated binaries

### PR DESCRIPTION
**Pull request by Disservin**

### CODE REVIEW

**Comment by vondele** [comment]
This looks good, I wonder if we could ensure that for all architectures we generate in CI, we also run some testing, at least verify the bench.

**Comment by cj5716** [comment]
I suspect the fail is because of no bench reference

**Comment by ppigazzini** [comment]
I test my CI patches on my Github account, using for the patch one of the branches that triggers the CI, or temporary adding the branch name in the list.

https://github.com/official-stockfish/Stockfish/blob/9a2d50ecccfc737249245280586924ee3ef53abb/.github/workflows/stockfish.yml#L2-L13

**Comment by Disservin** [comment]
> I suspect the fail is because of no bench reference

no, apply tried to compile for avx2/pext that why it failed, if i see it correctly.

**Comment by ppigazzini** [comment]
> I suspect the fail is because of no bench reference

CPU without arch extension, the bench reference and test of modern ARCH successfully tested.

> No signature obtained from bench. Code crashed or assert triggered ?

**Comment by Disservin** [comment]
> I test my CI patches on my Github account, using for the patch one of the branches that triggers the CI, or temporary adding the branch name in the list.
> 
> https://github.com/official-stockfish/Stockfish/blob/9a2d50ecccfc737249245280586924ee3ef53abb/.github/workflows/stockfish.yml#L2-L13

ty, that's what i did in the past though, I though that this would be rather straight forward ; )

**Comment by Disservin** [comment]
Now, what I dont understand is why it tries to run the bmi2 step for `Android NDK arm`

**Comment by ppigazzini** [comment]
`stockfish_test.yml` seems fine, Github fluke?

**Comment by Disservin** [comment]
idk for some reason `if: ${{ matrix.config.run_64bit_tests }} && runner.os != 'macOS'` is evaluated as true for android which i dont get because `run_64bit_tests` isnt defined... i got the same results in my own branch...

**Comment by ppigazzini** [comment]
Same error on my repo for that branch...

**Comment by ppigazzini** [comment]
> idk for some reason `if: ${{ matrix.config.run_64bit_tests }} && runner.os != 'macOS'` is evaluated as true for android which i dont get because `run_64bit_tests` isnt defined... i got the same results in my own branch...

${{ runner.os }}

File: .github/workflows/stockfish_test.yml [hunk 1]
```diff
@@ -179,6 +179,20 @@ jobs:
           make -j2 ARCH=x86-64-modern optimize=no debug=yes build
           ../tests/signature.sh $benchref
 
+      - name: Test x86-64-bmi2 build
+        if: ${{ matrix.config.run_64bit_tests }} && runner.os != 'macOS'
```
Comment (ppigazzini) [inline .github/workflows/stockfish_test.yml:183]: && ${{ runner.os }}

**Comment by ppigazzini** [comment]
Just as reference, it seems that the braces must enclose the whole expression. It worked in that way on my repo.
https://docs.github.com/en/actions/learn-github-actions/expressions#example-expression-in-an-if-conditional

```diff
diff --git a/.github/workflows/stockfish_test.yml b/.github/workflows/stockfish_test.yml
index 8a71d76b..b5866314 100644
--- a/.github/workflows/stockfish_test.yml
+++ b/.github/workflows/stockfish_test.yml
@@ -179,6 +179,21 @@ jobs:
           make -j2 ARCH=x86-64-modern optimize=no debug=yes build
           ../tests/signature.sh $benchref

+      - name: Test x86-64-bmi2 build
+        if: ${{ matrix.config.run_64bit_tests && runner.os != 'macOS' }}
+        run: |
+          echo "Matrix variables: ${{ toJSON(matrix) }}"
+          make clean
+          make -j2 ARCH=x86-64-bmi2 build
+          ../tests/signature.sh $benchref
+
+      - name: Test x86-64-avx2 build
+        if: ${{ matrix.config.run_64bit_tests && runner.os != 'macOS' }}
+        run: |
+          make clean
+          make -j2 ARCH=x86-64-avx2 build
+          ../tests/signature.sh $benchref
+
       - name: Test x86-64-modern build
         if: ${{ matrix.config.run_64bit_tests }}
         run: |
```

**Pull request closed without merging**